### PR TITLE
Delete G-Code File on Octoprint

### PR DESCRIPTION
**Pull request by jreed1701**
Call back added should trigger a web socket connection back to
octoprint TSD plugin to call delete function.

There is a "sister" update in the TSD Octoprint plugin repo, and the PR link is noted below.

Work In Progress. Issues to figure out:
* When clicking the delete button, need to figure out how to get a web connection to each printer in GCodePage.vue. Right now, the connection is null because I don't have a printer id.

Questions:
* When deleting GCode, should it be deleted on all printers? Or, a specfic printer? Can't assume only uploaded to one.

Please Note: Started work off of "release" branch because "master" was not working at the time. At some point, will need to change the base of this branch over to master.

Relates to: https://github.com/TheSpaghettiDetective/OctoPrint-TheSpaghettiDetective/pull/159
Relates to: https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/issues/601

### CODE REVIEW

**Comment by jreed1701** [comment]
My current goal is to click the delete button and see that I triggered the delete function on the octoprint plugin via a log file.  Right now, I can click the delete button and get the callback in GCodePage.vue to fire but I'm having trouble getting a web socket to the octoprint plugin going.  

Here's a link to the code snippet I refer to:
https://github.com/jreed1701/TheSpaghettiDetective/blob/fc938275f1ee13ec12580f4b087244186a5dbd21/web/frontend/src/printers/GCodesPage.vue#L338-L408

File: docker-compose.yml [hunk 1]
```diff
@@ -10,24 +11,23 @@ x-web-defaults: &web-defaults
...
     SOCIAL_LOGIN: 'False'
     REDIS_URL: 'redis://redis:6379'
     DATABASE_URL: 'sqlite:////app/db.sqlite3'
     INTERNAL_MEDIA_HOST: 'http://web:3334'
     ML_API_HOST: 'http://ml_api:3333'
-    ACCOUNT_ALLOW_SIGN_UP: 'False'  # -> set to 'True' if you want to open sign up form
+    ACCOUNT_ALLOW_SIGN_UP: 'False' # -> set to 'True' if you want to open sign up form
     WEBPACK_LOADER_ENABLED:
     OCTOPRINT_TUNNEL_PORT_RANGE: '${OCTOPRINT_TUNNEL_PORT_RANGE-15853-15873}'
-
```
Comment (kennethjiang) [inline docker-compose.yml:30]: Same here.

File: docker-compose.yml [hunk 2]
```diff
@@ -44,21 +44,21 @@ services:
     build:
       context: ml_api
     environment:
-        DEBUG: 'True'
```
Comment (kennethjiang) [inline docker-compose.yml:47]: Same here.

File: docker-compose.yml [hunk 3]
```diff
@@ -44,21 +44,21 @@ services:
...
+      #ML_API_TOKEN:
+      #HAS_GPU: 'False'
     command: bash -c "gunicorn --bind 0.0.0.0:3333 --workers 1 wsgi"
 
   web:
     <<: *web-defaults
     hostname: web
     ports:
-      - "3334:3334"
+      - "3000:3334"
```
Comment (kennethjiang) [inline docker-compose.yml:57]: Any reason why you want to change the port?

File: web/frontend/src/lib/printer_comm.js [hunk 4]
```diff
@@ -113,6 +113,9 @@ export default function PrinterComm(printerId, wsUri, onPrinterUpdateReceived, o
   }
 
   self.passThruToPrinter = function(msg, callback) {
+    console.log('**************************************************')
```
Comment (kennethjiang) [inline web/frontend/src/lib/printer_comm.js:116]: Logging like this is considered a bad practice and should be removed.

**Review by kennethjiang — COMMENTED** [review]
This PR needs some cleanup before it can be reviewed. We don't have a formal code guidelines but our codes should adhere to generally-accepted good practices like this one: https://gist.github.com/luismts/495d982e8c5b1a0ced4a57cf3d93cf60

**Comment by encetamasb** [comment]
Hi! 

Brave first task :)

You should know that file handling is going to change sooner or later. I don't know what's the planned approach exactly, but as far as I know, a better way of "managing remote files" is on the table.

"When deleting GCode, should it be deleted on all printers? Or, a specfic printer? Can't assume only uploaded to one."

Gcodes are not uploaded to printers automatically. I would assume that it gets deleted everywhere if I click "the" button. IMO we need another delete button for this, like delete from tsd, delete from tsd and printers... 

And in general, we will need to be double-check every code paths what may delete files (or everything) on printers, so expect strict reviews around those parts. :)

"When clicking the delete button, need to figure out how to get a web connection to each printer in GCodePage.vue. Right now, the connection is null because I don't have a printer id."

I would send the delete request to the server, and then send the command to all/some printers there. No need to handle this frontend side in my opinion.

"Please Note: Started work off of "release" branch because "master" was not working at the time. At some point, will need to change the base of this branch over to master."

Possibly you need to build frontend first. `cd frontend && yarn && yarn build`
Don't commit updated files in builds dir please, for now.

**Comment by jreed1701** [comment]
@kennethjiang - This should be a draft. Tried to make it very clear in chat that I was no where near finished with the feature.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,6 +1,7 @@
 version: '2.4'
 
-x-web-defaults: &web-defaults
+x-web-defaults:
```
[the remark this answers is not in the document]
Comment (jreed1701) [reply docker-compose.yml]: Please ignore all changes to docker-compose.yml.  I am removing them.

**Comment by kennethjiang** [comment]
@jreed1701 Are you still working on this PR? If not, I'll close it as it's probably become too outdated to merge anyway.

**Pull request closed without merging**